src/read.c

```diff
@@ -4710,18 +4732,24 @@ static avifResult avifMetaFindAlphaItem(avifMeta * meta,
         result = avifMetaFindOrCreateItem(meta, newItemID, alphaItem); // Create new empty item.
     }
     if (result != AVIF_RESULT_OK) {
-        avifFree(alphaItemIndices);
+        avifFree(dimgIdxToAlphaItemIdx);
         *isAlphaItemInInput = AVIF_FALSE;
         return result;
     }
     memcpy((*alphaItem)->type, "grid", 4); // Make it a grid and register alpha items as its tiles.
     (*alphaItem)->width = colorItem->width;
     (*alphaItem)->height = colorItem->height;
-    for (uint32_t i = 0; i < alphaItemCount; ++i) {
-        avifDecoderItem * item = meta->items.item[alphaItemIndices[i]];
-        item->dimgForID = (*alphaItem)->id;
+    for (uint32_t dimgIdx = 0; dimgIdx < tileCount; ++dimgIdx) {
+        if (dimgIdxToAlphaItemIdx[dimgIdx] == itemIndexNotSet) {
+            avifFree(dimgIdxToAlphaItemIdx);
+            *isAlphaItemInInput = AVIF_FALSE;
```

Comment:
It is not cleae whether we should add `*alphaItem = NULL;` here. We can avoid this issue by performing this check early as I suggested in the comment at line 4701.